src/pages/patientView/mutation/VAFLineChart.tsx

```diff
@@ -0,0 +1,773 @@
+
+        const highlightedMutations = [];
+        if (!this.props.dataStore.onlyShowSelectedInVAFChart) {
+            // dont bold highlighted mutations if we're only showing highlighted mutations
+            highlightedMutations.push(...this.props.dataStore.selectedMutations);
+        }
+        const mouseOverMutation = this.props.dataStore.getMouseOverMutation();
+        if (mouseOverMutation) {
+            highlightedMutations.push(mouseOverMutation);
+        }
+        if (highlightedMutations.length > 0 && this.scale !== null && this.highlightContainer !== null) {
+            return highlightedMutations.map(highlightedMutation=>{
+                const points = this.mutationToDataPoints.get({
+                    proteinChange: highlightedMutation.proteinChange,
+                    hugoGeneSymbol: highlightedMutation.gene.hugoGeneSymbol
+                });
+                if (!points) {
+                    return <g/>
+                }
+                let linePath = null;
+                if (points.length > 1) {
+                    // more than one point -> we should render a path
+                    let d = `M ${this.scale!.x(points[0].x)} ${this.scale!.y(this.y(points[0]))}`;
+                    for (let i=1; i<points.length; i++) {
+                        d = `${d} L ${this.scale!.x(points[i].x)} ${this.scale!.y(this.y(points[i]))}`;
+                    }
+                    linePath = (
+                        <path
+                            style={{
+                                stroke:HIGHLIGHT_COLOR,
+                                strokeOpacity:1,
+                                strokeWidth:HIGHLIGHT_LINE_STROKE_WIDTH,
+                                fillOpacity:0,
+                                pointerEvents:"none"
+                            }}
+                            d={d}
+                        />
+                    );
+                }
+                const pointPaths = points.map(point=>(
+                    <path
+                        d={
+                            `M ${this.scale!.x(point.x)} ${this.scale!.y(this.y(point))}
+                            m -${SCATTER_DATA_POINT_SIZE}, 0
+                            a ${SCATTER_DATA_POINT_SIZE}, ${SCATTER_DATA_POINT_SIZE} 0 1,0 ${2*SCATTER_DATA_POINT_SIZE},0
+                            a ${SCATTER_DATA_POINT_SIZE}, ${SCATTER_DATA_POINT_SIZE} 0 1,0 ${-2*SCATTER_DATA_POINT_SIZE},0
+                            `
+                        }
+                        style={{
+                            stroke:HIGHLIGHT_COLOR,
+                            fill:"white",
+                            strokeWidth:2,
+                            opacity:1,
+                            pointerEvents:"none"
+                        }}
+                    />
+                ));
+
+                return (
+                    <Portal isOpened={true} node={this.highlightContainer}>
```

Comment:
curious why this is in a portal

Reply:
ah its legacy from when I had to (1) put those in an observer (2) put those somewhere inside the victory component due to layering. but now since it just goes on the top it's not necessary. I'll simplify.